vertices_to_h5m/core.py

```diff
@@ -151,24 +156,228 @@ def vertices_to_h5m(
+
+    # tag all nodes with -1
+    node_tag_data = np.full(len(vertices), -1)
+    node_tags = nodes_group.create_group("tags")
+    node_tags.create_dataset("GLOBAL_ID", data=node_tag_data)
+
+    elements = tstt.create_group("elements")
+
+    elems = {
+        "Edge": 1,
+        "Tri": 2,
+        "Quad": 3,
+        "Polygon": 4,
+        "Tet": 5,
+        "Pyramid": 6,
+        "Prism": 7,
+        "Knife": 8,
+        "Hex": 9,
+        "Polyhedron": 10,
+    }
+    tstt["elemtypes"] = h5py.enum_dtype(elems)
+
+    now = datetime.now()
+    tstt.create_dataset(
+        "history",
+        data=[
+            "vertices_to_h5m",
+            __version__,
+            now.strftime("%m/%d/%y"),
+            now.strftime("%H:%M:%S"),
+        ],
+    )
+
+    tri3_group = elements.create_group("Tri3")
+    tri3_group.attrs.create("element_type", elems["Tri"], dtype=tstt["elemtypes"])
+
+    connectivity_group = tri3_group.create_dataset(
+        "connectivity",
+        data=all_triangles + 1,  # node indices are 1 based in h5m
+        dtype=np.uint64,
+    )
+
+    connectivity_group.attrs.create("start_id", global_id)
+    global_id += len(all_triangles)
 
+    # give each local group a unique tag
+    tag_data = np.concatenate(
+        [np.full(len(group), i) for i, group in enumerate(local_triangle_groups)]
+    )
+    tags_tri3_group = tri3_group.create_group("tags")
+    tags_tri3_group.create_dataset("GLOBAL_ID", data=tag_data)
+
+    tags_tstt_group = tstt.create_group("tags")
+
+    cat_group = tags_tstt_group.create_group("CATEGORY")
+    cat_group.attrs.create("class", 1, dtype=np.int32)
+    id_list = [7, 8, 9]
+    cat_group.create_dataset(
+        "id_list",
+        data=id_list,
+        dtype=np.uint,
+    )
+    dt = h5py.opaque_dtype(np.dtype("S32"))
+    # dt.set_tag(b"")
+    cat_group["type"] = dt
+    cat_group.create_dataset(
+        "values",
+        data=["Surface", "Volume", "Group"],
+        dtype=cat_group["type"],
+    )
+
+    diri_group = tstt_tags_group.create_group("DIRICHLET_SET")
+    diri_group["type"] = np.dtype("i4")
+    diri_group.attrs.create("class", 1, dtype=np.int32)
+    diri_group.attrs.create("default", -1, dtype=diri_group["type"])
+    diri_group.attrs.create("global", -1, dtype=diri_group["type"])
+
+    geom_group = tstt_tags_group.create_group("GEOM_DIMENSION")
+    geom_group["type"] = np.dtype("i4")
+    geom_group.attrs.create("class", 1, dtype=np.int32)
+    geom_group.attrs.create("default", -1, dtype=geom_group["type"])
+    geom_group.attrs.create("global", -1, dtype=geom_group["type"])
+    geom_group.create_dataset("id_list", data=id_list, dtype=np.uint64)
+    geom_group.create_dataset("values", data=[2, 3, 4], dtype=geom_group["type"])
+
+    gs2_group = tags_tstt_group.create_group("GEOM_SENSE_2")
+    # TODO pymoab's data type is
+    # ```
+    # DATATYPE "type" H5T_ARRAY { [2] H5T_STD_U64LE };
+    # ```
+    # Not sure yet how to emulate it with h5py.
```

Comment:
Just wondering if @pshriwise can shed any light on emulating this MOAB datatype in h5py :pray: